code/game/gamemodes/miniantags/hades/hades.dm

```diff
@@ -0,0 +1,223 @@
+	..(AM,skipcatch,hitpush,blocked)
+	lastsinPerson -= 75
+
+/mob/living/simple_animal/hades/bullet_act(obj/item/projectile/P, def_zone)
+	..(P,def_zone)
+	lastsinPerson -= 75
+
+/mob/living/simple_animal/hades/attack_hand(mob/living/carbon/human/M)
+	..(M)
+	lastsinPerson -= 75
+
+/mob/living/simple_animal/hades/proc/Appear()
+	new /obj/effect/timestop(get_turf(src))
+
+/datum/objective/sinPersonholder
+	explanation_text = "You are feeling sinful."
+	dangerrating = 3
+
+/mob/living/simple_animal/hades/Life()
+	..()
+	if(world.time > lastsinPerson + sinPersonTime)
+		if(prob(fakesinPersonChance))
+			lastsinPerson = world.time
+			world << "<span class='warning'><font size=4>[pick(sinPersonsayings)]</font></span>"
+			//SUE ME
+			var/list/creepyasssounds = list('sound/effects/ghost.ogg', 'sound/effects/ghost2.ogg', 'sound/effects/Heart Beat.ogg', 'sound/effects/screech.ogg',\
+						'sound/hallucinations/behind_you1.ogg', 'sound/hallucinations/behind_you2.ogg', 'sound/hallucinations/far_noise.ogg', 'sound/hallucinations/growl1.ogg', 'sound/hallucinations/growl2.ogg',\
+						'sound/hallucinations/growl3.ogg', 'sound/hallucinations/im_here1.ogg', 'sound/hallucinations/im_here2.ogg', 'sound/hallucinations/i_see_you1.ogg', 'sound/hallucinations/i_see_you2.ogg',\
+						'sound/hallucinations/look_up1.ogg', 'sound/hallucinations/look_up2.ogg', 'sound/hallucinations/over_here1.ogg', 'sound/hallucinations/over_here2.ogg', 'sound/hallucinations/over_here3.ogg',\
+						'sound/hallucinations/turn_around1.ogg', 'sound/hallucinations/turn_around2.ogg', 'sound/hallucinations/veryfar_noise.ogg', 'sound/hallucinations/wail.ogg')
+			world << pick(creepyasssounds)
+		else
+			lastsinPerson = world.time
+			var/mob/living/carbon/human/sinPerson = pick(player_list)
+			if(sinPerson)
+				loc = get_turf(pick(oview(1,sinPerson)))
+				Appear()
+				var/sinPersonchoice = pick(list("Greed","Gluttony","Pride","Lust","Envy","Sloth","Wrath"))
+				if(sinPersonchoice == "Greed")
+					src.say("Your sin, [sinPerson], is Greed.")
+					if(prob(50))
+						src.say("I will indulge your sin, [sinPerson].")
+						var/list/greed = list(/obj/item/stack/sheet/mineral/gold,/obj/item/stack/sheet/mineral/silver,/obj/item/stack/sheet/mineral/diamond)
+						for(var/i = 0; i < 10; ++i)
+							var/greed_type = pick(greed)
+							new greed_type(get_turf(sinPerson))
+					else
+						src.say("Your sin will be punished, [sinPerson]!")
+						new/obj/structure/closet/statue(get_turf(sinPerson),sinPerson)
+				else if(sinPersonchoice == "Gluttony")
+					src.say("Your sin, [sinPerson], is Gluttony.")
+					if(prob(50))
+						src.say("I will indulge your sin, [sinPerson].")
+						var/list/allTypes = list()
+						for(var/A in typesof(/obj/item/weapon/reagent_containers/food/snacks))
+							var/obj/item/weapon/reagent_containers/food/snacks/O = A
+							if(initial(O.cooked_type))
+								allTypes += A
+						for(var/i = 0; i < 10; ++i)
+							var/greed_type = pick(allTypes)

```

Comment:
well while it won't lead to 0 (I read wrong, thought it was global), what's wrong with a rare chance to get 10 of the same food?